### ISSUE TITLE
Catch2 2.13.10 Compile error at 100% Android ARM64

### ISSUE THREAD

**Issue by defencedog** [catchorg/Catch2#2576]
What is this `_android_log_write` issue?

![SmartSelect_20221122_004932_Termux](https://user-images.githubusercontent.com/831749/203168046-f2fa15d4-4616-41f4-b38a-550570ad967b.jpg)

**Comment by horenmar** [comment]
Same question as on Discord: does the link step link `log` library?

**Comment by defencedog** [comment]
@horenmar Can you explain in bit more detail?
I have modified even this hardlink file location
```
nano /data/data/com.termux/files/home/Catch2-2.13.10/include/internal/catch_debug_console.cpp
#if defined(CATCH_CONFIG_ANDROID_LOGWRITE)
// #include <android/log.h>
#include "/data/data/com.termux/files/usr/include/android/log.h"
```

**Comment by defencedog** [comment]
@horenmar ~how can I use this flag https://github.com/catchorg/Catch2/issues/1743#issuecomment-528811088~
Ok I was unable to link `android/log.h` so I removed everything in `/data/data/com.termux/files/home/Catch2-2.13.10/include/internal/catch_debug_console.cpp` except the last function defining `namespace Catch`
Build is successful; but why it isn't linking to `android/log.h` ?

```
 *  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 *
 */

#include "catch_debug_console.h"
#include "catch_compiler_capabilities.h"
#include "catch_stream.h"
#include "catch_platform.h"
#include "catch_windows_h_proxy.h"
/*
#if defined(CATCH_CONFIG_ANDROID_LOGWRITE)
#include "/data/data/com.termux/files/usr/include/android/log.h"

    namespace Catch {
        void writeToDebugConsole( std::string const& text ) {
            __android_log_write( ANDROID_LOG_DEBUG, "Catch", text.c_str() );
        }
    }

#elif defined(CATCH_PLATFORM_WINDOWS)

    namespace Catch {
        void writeToDebugConsole( std::string const& text ) {
            ::OutputDebugStringA( text.c_str() );
        }
    }

#else */

    namespace Catch {
        void writeToDebugConsole( std::string const& text ) {
            // !TBD: Need a version for Mac/ XCode and other IDEs
            Catch::cout() << text;
        }
    }

//#endif // Platform
```

**Comment by horenmar** [comment]
Right the question isn't about whether you can include the file, but whether your link step links `log` or not. You can run `make VERBOSE=1` to get the command line used for the link step.

**Comment by horenmar** [comment]
And you can disable the logging support by configuring Catch2's build with `-DCATCH_CONFIG_ANDROID_LOGWRITE=OFF`.

**Comment by defencedog** [comment]
@horenmar If you guide me I would love to have this logging support because `reaktoro` engineering library needs it

**Comment by ThomasHezard** [comment]
Hello 👋    

I encountered the same issue when upgrading `Catch` (from a rather out-dated version... 🙄) in my repo.    
I solved it at first by adding `add_definitions("-DCATCH_CONFIG_NO_ANDROID_LOGWRITE")` for all my test targets, but I ended up with another solution.

`Catch` is integrated in my repo as a single header, with a custom-made `CMakeLists.txt` (maybe not the best practice, I'm considering changing this to use the `CMakeLists.txt` from `Catch` repo).    
Adding the following in my `CMakeLists.txt` solved this `undefined symbol: __android_log_write ` issue without having to change my test targets:
```cmake
if(ANDROID)
    message("-- Using Catch for Android: linking to log lib")
    find_library(log-lib log)
    target_link_libraries(Catch INTERFACE ${log-lib})
endif()
```
NB: the `if` statement is here because my repo builds for multiple platforms but you don't need that if you only build for Android.

This linking phase is actually [in the CMakeLists.txt from Catch repo](https://github.com/catchorg/Catch2/blob/v2.13.10/CMakeLists.txt#L101), so I guess that makes sense.    

Hope this helps 🙂
